### DISCUSSION TITLE
How to track all tables/relationships faster for existing db

### DISCUSSION THREAD

**Discussion by patrickkidd** [hasura/graphql-engine#7655] [Question] [answered]
I am just starting so I apologize if this is ignorant. Clicking "Track All" under "Untracked tables or views" in the UI takes a very long time (30 minutes) to run on a local test version of our production schema. Further, clicking "Track All" for "Untracked Foreign-key and relationships" takes just as long but doesn't seem to set up any fk relationships, showing one error popup for one relationship in the list:

```
[
    {
        "definition": {
            "using": {
                "foreign_key_constraint_on": {
                    "column": "org_id",
                    "table": {
                        "schema": "lasc0rp",
                        "name": "address_type"
                    }
                }
            },
            "name": "address_types",
            "source": "teamworks",
            "comment": null,
            "table": {
                "schema": "lasc0rp",
                "name": "school"
            }
        },
        "reason": "Inconsistent object: in table \"lasc0rp.school\": in relationship \"address_types\": more than one foreign key constraint exists on the given column(s)",
        "name": "array_relation address_types in table lasc0rp.school in source teamworks",
        "type": "array_relation"
    }
]
```

Is there a faster way to initialize the hasura metadata for this database? Troubleshooting proves quite difficult.

**Comment by BenoitRanque** [comment, marked as answer]
Assuming this is a known database & metadata, you can export the metadata from an existing project and import it into the new one.
The easiest way is accessed by clicking on the cog icon to the top right of the console
![image](https://user-images.githubusercontent.com/25712958/136473689-386277c2-5fce-4f45-bf35-7459c79978af.png)
You can then click on export/import metadata depending on whether this is a the existing or new project.

The above is however not the recommended way if you need to version control your metadata.
For that, you would need to use the [hasura cli](https://hasura.io/docs/latest/graphql/core/hasura-cli/index.html) to export or apply metadata. Check out the docs [here](https://hasura.io/docs/latest/graphql/core/migrations/manage-metadata.html)

This is all assuming you have an existing hasura project where the tables and relations are already tracked

**Comment by patrickkidd** [comment]
OK, gotcha. So this just needs to take a while the first time but then you can just import the metadata after that. For example when spinning up a new instance in a CI/CD system I assume this import reasonably fast from a file.

**Reply by BenoitRanque** [reply]
That is correct.

Do note that not tracking all tables automatically is an intentional design decision of hasura.
There are many cases where you may not want to expose all tables to the graphql api, especially when working with existing, large databases

**Comment by patrickkidd** [comment]
makes perfect sense. thank you
